core/src/main/java/io/questdb/griffin/engine/functions/catalogue/PgGetKeywordsFunctionFactory.java

```diff
@@ -0,0 +1,129 @@
+ *  limitations under the License.
+ *
+ ******************************************************************************/
+
+package io.questdb.griffin.engine.functions.catalogue;
+
+import io.questdb.cairo.CairoConfiguration;
+import io.questdb.cairo.ColumnType;
+import io.questdb.cairo.GenericRecordMetadata;
+import io.questdb.cairo.TableColumnMetadata;
+import io.questdb.cairo.sql.Function;
+import io.questdb.cairo.sql.NoRandomAccessRecordCursor;
+import io.questdb.cairo.sql.Record;
+import io.questdb.cairo.sql.RecordMetadata;
+import io.questdb.griffin.FunctionFactory;
+import io.questdb.griffin.SqlExecutionContext;
+import io.questdb.griffin.engine.functions.CursorFunction;
+import io.questdb.griffin.engine.functions.GenericRecordCursorFactory;
+import io.questdb.std.IntList;
+import io.questdb.std.ObjList;
+
+public class PgGetKeywordsFunctionFactory implements FunctionFactory {
+    private static final RecordMetadata METADATA;
+
+    @Override
+    public String getSignature() {
+        return "pg_get_keywords()";
+    }
+
+    @Override
+    public boolean isRuntimeConstant() {
+        return true;
+    }
+
+    @Override
+    public Function newInstance(int position, ObjList<Function> args, IntList argPositions, CairoConfiguration configuration, SqlExecutionContext sqlExecutionContext) {
+        return new CursorFunction(
+                new GenericRecordCursorFactory(METADATA, new KeywordsCatalogueCursor(), false)
+        ) {
+            @Override
+            public boolean isRuntimeConstant() {
+                return true;
+            }
+        };
+    }
+
+    private static class KeywordsCatalogueCursor implements NoRandomAccessRecordCursor {
+        private static final int rowCount = Constants.KEYWORDS.length;
+        private final KeywordCatalogueRecord record = new KeywordCatalogueRecord();
+        private int row = -1;
+
+        @Override
+        public void close() {
+            row = -1;
+        }
+
+        @Override
+        public Record getRecord() {
+            return record;
+        }
+
+        @Override
+        public boolean hasNext() {
+            return ++row < rowCount;
+        }
+
+        @Override
+        public long size() {
+            return rowCount;
+        }
+
+        @Override
+        public void toTop() {
+            row = -1;
+        }
+
+        class KeywordCatalogueRecord implements Record {
+
+            @Override
+            public CharSequence getStr(int col) {
+                if (col == 0) {
+                    return Constants.KEYWORDS[row];
+                }
+                return null;
+            }
+
+            @Override
+            public CharSequence getStrB(int col) {
+                return getStr(col);
+            }
+
+            @Override
+            public int getStrLen(int col) {
+                if (col == 0) {
+                    return getStr(col).length();
+                }
+                return -1;
+            }
+        }
+    }
+
```

Comment:
Shouldn't function return more columns ? 
pg_get_keywords () → setof record ( word text, catcode "char", barelabel boolean, catdesc text, baredesc text )
see https://www.postgresql.org/docs/current/functions-info.html

Reply:
That's not a new function, but an old one. Added missing columns in 4ff3df82553c4c551956cbdcf551cf215a5eb778